File: backend/src/main/java/com/artifax_solutions/website/model/Pricing.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Pricing {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String subTitle;
    private String title;
    private String description;
    @OneToMany(mappedBy = "pricing")
    private List<PricingCard> cardList;
}

File: frontend/src/app/models/Pricing.ts
export interface Pricing {
  id: number | undefined | null;
  subTitle: string | undefined | null;
  title: string | undefined | null;
  description: string | undefined | null;
  cardList: PricingCard[] | undefined | null;
}

export interface PricingCard {
  id: number | undefined | null;
  title: string | undefined | null;
  price: string | undefined | null;
  description: string | undefined | null;
  pricingServiceList: PricingService[] | undefined | null;
  buttonText: string | undefined | null;
  buttonLink: string | undefined | null;
}

export interface PricingService {
  id: number | undefined | null;
  icon: string | undefined | null;
  benefit: string | undefined | null;
}
